### PR TITLE
fix(iot-e2e): add build/test scripts for Android Things

### PR DESCRIPTION
**Pull request by jasmineymlo**
add build scripts required for nightly build and running end-to-end test on Android Things

### CODE REVIEW

File: iot-e2e-tests/android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-bin.zip
 zipStoreBase=GRADLE_USER_HOME
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/gradle/wrapper/gradle-wrapper.properties]: Are these changes same for both Android and things?

File: iot-e2e-tests/android/runInstrumentationTests.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import subprocess
+import os
+import time
+import sys
+
+env=dict(os.environ, PATH="path")
+#cmd = "adb -s "+sys.argv[1]+" shell am instrument -w -r -e debug false -e IOTHUB_CONNECTION_STRING '"+env["IOTHUB_CONNECTION_STRING"]+"' -e IOTHUB_E2E_X509_CERT_BASE64 "+env["IOTHUB_E2E_X509_CERT_BASE64"]+" -e IOTHUB_E2E_X509_PK_BASE64 "+env["IOTHUB_E2E_X509_PRIVATE_KEY_BASE64"]+" -e IOTHUB_E2E_X509_THUMBPRINT "+env["IOTHUB_E2E_X509_THUMBPRINT"]+"  -e package com.microsoft.azure.sdk.iot.android com.iothub.azure.microsoft.com.androidsample.test/android.support.test.runner.AndroidJUnitRunner"
+cmd = "adb -s "+sys.argv[1]+" shell am instrument -w -r -e debug false -e package 'com.microsoft.azure.sdk.iot.androidthings' com.microsoft.azure.sdk.iot.androidthings.test/android.support.test.runner.AndroidJUnitRunner"
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/runInstrumentationTests.py]: Can you take care of what is commented here? And only reflect what's relevant

File: iot-e2e-tests/android/runInstrumentationTests.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+print(cmd)
+p = subprocess.Popen(cmd,  stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf8')
+(out, err) = p.communicate()
+print(out)
+print(err)
+os.popen("taskkill /IM cmd.exe /FI \"WINDOWTITLE eq C:\windows\system32\cmd.exe - emulator*\"")
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/runInstrumentationTests.py]: Make sure your print stmnts are more descriptive to understand during debugging

File: iot-e2e-tests/android/things/build.gradle [hunk 4]
```diff
@@ -71,7 +71,7 @@ dependencies {
     implementation 'com.android.support.constraint:constraint-layout:1.1.3'
     implementation 'com.android.support:multidex:1.0.3'
 
-    implementation ('com.microsoft.azure.sdk.iot:iot-e2e-common:0.20.2'){
+    implementation ('com.microsoft.azure.sdk.iot:iot-e2e-common:0.21.0'){
         exclude module: 'junit'
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/things/build.gradle]: Why was this not incremented during release?

Comment (jasmineymlo) [reply iot-e2e-tests/android/things/build.gradle]: will need to update bump version script from internal repo

File: iot-e2e-tests/android/runInstrumentationTests.py (hunk 3 above)
Comment (jasmineymlo) [reply iot-e2e-tests/android/runInstrumentationTests.py]: currently, we are capturing whatever is printed on the testlog which is the same as what we have when we are testing with AppCenter.

File: iot-e2e-tests/android/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (jasmineymlo) [reply iot-e2e-tests/android/gradle/wrapper/gradle-wrapper.properties]: yes, only upgrading gradle version.

**Pull request merged**